internal/cli/opsmanager/serverusage/download.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverusage
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+type DownloadOpts struct {
+	cli.DownloaderOpts
+	startDate string
+	endDate   string
+	format    string
+	store     store.ServerUsageReportDownloader
+}
+
+var downloadMessage = "Download of %s completed.\n"
+
+func (opts *DownloadOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DownloadOpts) Run() error {
+	f, err := opts.NewWriteCloser()
+	if err != nil {
+		return err
+	}
+
+	if err := opts.store.DownloadServerUsageReport(opts.newServerTypeOptions(), f); err != nil {
+		_ = opts.OnError(f)
+		return err
+	}
+	if opts.Out != "/dev/stdout" {
+		fmt.Printf(downloadMessage, opts.Out)
+	}
+	return f.Close()
+}
+
+func (opts *DownloadOpts) newServerTypeOptions() *opsmngr.ServerTypeOptions {
+	return &opsmngr.ServerTypeOptions{
+		StartDate:  opts.startDate,
+		EndDate:    opts.endDate,
+		FileFormat: opts.format,
+	}
+}
+
+func (opts *DownloadOpts) initDefaultOut() {
+	opts.format = "." + opts.format
```

Comment:
for zip to work is has to be exactly `zip` with no dots, anything else the API assumes tar.gz

Reply:
I may have misunderstood your comment here https://github.com/mongodb/mongocli/pull/501#discussion_r507611698.
The behaviour now is that the user has to specify the format flag as`--format zip` and `--format tar.gz`. 
Before it was `--format .zip` and `--format .tar.gz`

Reply:
mongocli use `tar.gz` if the flag is not specified

Reply:
when `opts.format = zip`, `opts.Out = "serverUsage-startDate-endDate.zip"`. Am I missing something?

Reply:
yes, the behaviour is correct, but you're adding a dot to the query param, so the server will get `.zip` which will not match the check the server does `"zip".equals(param)` and then defaulting to tar.gz

Reply:
got it, Thanks for the explanation

Reply:
this is not about the file name, but what you're sending to the server, adding it here solves the file name but breaks the API request

Reply:
I wrote this message https://github.com/mongodb/mongocli/pull/501#discussion_r507661018 before reading your comment (I didn’t refresh the page)

Reply:
I am not sure why I thought that I need to pass ".zip" and ".tar.gz" to the api 🤷‍♂️